mqtt/mqtt-bridge/src/persist/loader.rs

```diff
@@ -77,45 +77,125 @@ where
 
             // refresh next batch
             // if error, either someone forged the database or we have a database schema change
-            let next_batch = self.next_batch()?;
-            let mut inner = self.0.lock();
-            inner.batch = next_batch;
-
-            // get next element and return it
-            let maybe_extracted = inner.batch.pop_front();
-            let mut state_lock = inner.state.lock();
-            maybe_extracted.map_or_else(
-                || {
-                    state_lock.set_waker(cx.waker());
-                    Poll::Pending
-                },
-                |extracted| Poll::Ready(Some(Ok(extracted))),
-            )
+            let poll;
+            {
+                let batch_future = self.next_batch();
+                let mut batch_future = Box::pin(batch_future);
+                poll = batch_future.as_mut().poll(cx);
+            }
+            println!("poll {:?}", poll);
```

Comment:
Currently debugging a bug I noticed while testing, so feel free to ignore these printlns